wandb/sdk/wandb_run.py

```diff
@@ -3575,9 +3575,14 @@ def __getattr__(self, item: str) -> Any:
         self._assert_instance()
         return getattr(self._instance, item)
 
-    def wait(self) -> ArtifactInterface:
+    def wait(self, wait_timeout_secs: int = None) -> ArtifactInterface:
         if not self._instance:
-            resp = self._future.get().response.log_artifact_response
+            future_get = self._future.get(wait_timeout_secs)
+            if not future_get:
+                raise TimeoutError(
```

Comment:
Could we define our own `TimeoutError` type instead? From [the docs for the stdlib's TimeoutError](https://docs.python.org/3/library/exceptions.html#TimeoutError), it sounds like it's meant to represent low-level system errors, rather than issues in application logic.

Reply:
Hmm, do we have another error we raise when existing timeouts elapse? I would be surprised if we didn't already have this exact logic somewhere else.

Reply:
Is there a place to see all the wandb.errors? otherwise I just defined a **WaitTimeoutError**. Most timeout related errors either return ValueErrors or TimeoutErrors from what I've gathered after quickly searching around in this repo